libraries/botbuilder-dialogs-adaptive-testing/src/testActions/memoryAssertions.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Expression } from 'adaptive-expressions';
+import { TestAdapter, TurnContext } from 'botbuilder-core';
+import { Inspector, TestAction } from '../testAction';
+
+export interface MemoryAssertionsConfiguration {
+    description?: string;
+    assertions?: string[];
+}
+
+/**
+ * Run assertions against memory.
+ */
+export class MemoryAssertions extends TestAction implements MemoryAssertionsConfiguration {
+    public static $kind = 'Microsoft.Test.MemoryAssertions';
+
+    /**
+     * Gets or sets the description of this assertion.
+     */
+    public description: string;
+
+    /**
+     * Gets or sets the assertions.
+     */
+    public assertions: string[] = [];
+
+    /**
+     * Execute the test.
+     * @param testAdapter Adapter to execute against.
+     * @param callback Logic for the bot to use.
+     * @param inspector Inspector for dialog context.
+     * @returns A Promise that represents the work queued to execute.
+     */
+    public async execute(
+        adapter: TestAdapter,
+        callback: (context: TurnContext) => Promise<void>,
+        inspector?: Inspector
+    ): Promise<void> {
+        if (inspector) {
```

Comment:
Is it an error case to use this assertion without passing in an inspector? Perhaps an error should be thrown in an else block.

Reply:
In C#, there is a trace info in the else block:
`Trace.TraceInformation($"[Turn Ended => No inspector for {Description} MemoryAssertions]");`
I'm not sure if we should throw errors if there is no inspector although the memory assertion requires inspector to run.